src/Explorer/Panes/PublishNotebookPaneComponent.tsx

```diff
@@ -0,0 +1,209 @@
+import { FileSystemUtil } from "../Notebook/FileSystemUtil";
+import "./PublishNotebookPaneComponent.less";
+
+export interface PublishNotebookPaneProps {
+  notebookName: string;
+  notebookDescription: string;
+  notebookTags: string;
+  notebookAuthor: string;
+  onChangeDescription: (newValue: string) => void;
+  onChangeTags: (newValue: string) => void;
+  onChangeImageSrc: (newValue: string) => void;
+  onError: (formError: string, formErrorDetail: string, area: string) => void;
+  clearFormError: () => void;
+}
+
+interface PublishNotebookPaneState {
+  type: string;
+  notebookDescription: string;
+  notebookTags: string;
+  imageSrc: string;
+}
+
+export class PublishNotebookPaneComponent extends React.Component<PublishNotebookPaneProps, PublishNotebookPaneState> {
+  private readonly maxImageSizeInMib = 1.5;
+  private readonly ImageTypes = ["URL", "Custom Image"];
+  private descriptionPara1: string;
+  private descriptionPara2: string;
+  private descriptionProps: ITextFieldProps;
+  private tagsProps: ITextFieldProps;
+  private thumbnailUrlProps: ITextFieldProps;
+  private thumbnailSelectorProps: IDropdownProps;
+  private dateString: string;
+  private imageToBase64: (file: File, updateImageSrc: (result: string) => void) => void;
+
+  constructor(props: PublishNotebookPaneProps) {
+    super(props);
+
+    this.state = {
+      type: this.ImageTypes[0],
+      notebookDescription: "",
+      notebookTags: "",
+      imageSrc: undefined
+    };
+
+    this.imageToBase64 = (file: File, updateImageSrc: (result: string) => void) => {
+      const reader = new FileReader();
+      reader.readAsDataURL(file);
+      reader.onload = function() {
+        updateImageSrc(reader.result.toString());
+      };
+
+      const onError = this.props.onError;
+      reader.onerror = function(error) {
+        const formError = `Failed to convert ${file.name} to base64 format`;
+        const formErrorDetail = `${error}`;
+        const area = "PublishNotebookPaneComponent/selectImageFile";
+        onError(formError, formErrorDetail, area);
+      };
+    };
+
+    this.descriptionPara1 =
+      "This notebook has your data. Please make sure you delete any sensitive data/output before publishing.";
+
+    this.descriptionPara2 = `Would you like to publish and share ${FileSystemUtil.stripExtension(
+      this.props.notebookName,
+      "ipynb"
+    )} to the gallery?`;
+
+    this.thumbnailUrlProps = {
+      label: "Cover image url",
+      ariaLabel: "Cover image url",
+      onChange: (event, newValue) => {
+        this.props.onChangeImageSrc(newValue);
+        this.setState({ imageSrc: newValue });
+      }
+    };
+
+    this.thumbnailSelectorProps = {
+      label: "Cover image",
+      defaultSelectedKey: this.ImageTypes[0],
+      ariaLabel: "Cover image",
+      options: this.ImageTypes.map((value: string) => {
+        return { text: value, key: value };
```

Comment:
nit: FYI, you can also do: `this.ImageTypes.map((value: string) => ({ text: value, key: value })`